client/src/main/java/ChatApp.java

```diff
@@ -1,31 +1,23 @@
 import javafx.application.Application;
 import javafx.fxml.FXMLLoader;
 import javafx.scene.Parent;
-import javafx.scene.Scene;
 import javafx.stage.Stage;
 
-import java.io.IOException;
-
 /**
  * Сущность чата.
  *
  * @author Kristina Retivykh
  */
 public class ChatApp extends Application {
+
     @Override
     public void start(Stage primaryStage) throws Exception {
-        Network network = Network.getInstance();
-        Parent root = FXMLLoader.load(getClass().getResource("chat.fxml"));
-        primaryStage.setScene(new Scene(root));
-        primaryStage.setTitle("Chat");
-        primaryStage.setResizable(false);
-        primaryStage.show();
-        primaryStage.setOnCloseRequest(request -> {
-            try {
-                network.writeMessage("/quit");
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
-        });
+        MainController mainController = new MainController(primaryStage);
+        Parent login = FXMLLoader.load(getClass().getResource("login.fxml"));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource).
Calling this.getClass().getResource(...) could give results other than expected if this class is extended by a class in another package.